jupyterhub/jupyterhub/base/traefik-deployment.yaml

```diff
@@ -0,0 +1,102 @@
+---
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: traefik-proxy
+  labels:
+    k8s-app: traefik-proxy
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      k8s-app: traefik-proxy
+  strategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        k8s-app: traefik-proxy
+        name: traefik-proxy
+    spec:
+      serviceAccountName: traefik-proxy
+
+      volumes:
+      - name: config
+        configMap:
+          name: traefik-config
+      - name: watched
+        emptyDir: {}
+
+      initContainers:
+      - name: init-touch-rules
+        image: "registry.access.redhat.com/ubi8/ubi-micro:8.4"
+        volumeMounts:
+        - mountPath: "/watched"
+          name: "watched"
+        command:
+        - "/bin/bash"
+        - -c
+        args:
+        # this file has to exist before traefik starts
+        # so the watch can be set up properly
+        - touch /watched/rules.toml
+
+      containers:
+      - image: quay.io/internaldatahub/traefik:v1.7-build
```

Comment:
@tmckayus I'm guessing this isn't the productized image here.  Will you need to add another magic sed at build-time to grab the right image?

Reply:
Yes! good point.  We'll need the CPaaS build of traefik (quay.io/modh/odh-traefik-container)